search/views.py

```diff
@@ -11,13 +11,114 @@
 from django.views.decorators.http import require_POST
 
 from eventtracking import tracker as track
-from .api import perform_search
+from .api import perform_search, course_discovery_search
 
 # log appears to be standard name used for logger
 log = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
 @require_POST
+def course_discovery(request):
+    """
+    Search for courses
+
+    Args:
+        request (required) - django request object
+
+    Returns:
+        http json response with the following fields
+            "took" - how many seconds the operation took
+            "total" - how many results were found
+            "max_score" - maximum score from these resutls
+            "results" - json array of result documents
+
+            or
+
+            "error" - displayable information about an error that occured on the server
+
+    POST Params:
+        "search_string" (optional) - text with which to search for courses
+        "page_size" (optional)- how many results to return per page (defaults to 20, with maximum cutoff at 100)
+        "page_index" (optional) - for which page (zero-indexed) to include results (defaults to 0)
+    """
+    results = {
+        "error": _("Nothing to search")
+    }
+    status_code = 500
+
+    try:
```

Comment:
In general I think it's better to wrap your try/except protection as tightly as possible around the piece of code you're concerned with, versus doing a lot of different things within the try block.